diff --git a/build/gulpfile.vscode.ts b/build/gulpfile.vscode.ts
index d1475b73..6ddd3fa6 100644
--- a/build/gulpfile.vscode.ts
+++ b/build/gulpfile.vscode.ts
@@ -670,6 +670,11 @@ BUILD_TARGETS.forEach(buildTarget => {
 		gulp.task(vscodeTaskCI);
 
 		let vscodeTask: task.Task;
+		const packingTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}-packing`, task.series(
+			vscodeTaskCI
+		));
+		gulp.task(packingTask);
+
 		if (useEsbuildTranspile) {
 			const esbuildBundleTask = task.define(
 				`esbuild-bundle${dashed(platform)}${dashed(arch)}${dashed(minified)}`,
@@ -681,7 +686,7 @@ BUILD_TARGETS.forEach(buildTarget => {
 					minified && useCdnSourceMapsForPackagingTasks ? `${sourceMappingURLBase}/core` : undefined
 				)
 			);
-			vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+			const prepackTask = task.define(`vscode${dashed(minified)}-prepack`, task.series(
 				copyCodiconsTask,
 				cleanExtensionsBuildTask,
 				compileNonNativeExtensionsBuildTask,
@@ -689,17 +694,25 @@ BUILD_TARGETS.forEach(buildTarget => {
 				compileExtensionMediaBuildTask,
 				writeISODate('out-build'),
 				esbuildBundleTask,
-				vscodeTaskCI
 			));
-		} else {
+			gulp.task(prepackTask);
 			vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+				prepackTask,
+				packingTask,
+			));
+		} else {
+			const prepackTask = task.define(`vscode${dashed(minified)}-prepack`, task.series(
 				minified ? compileBuildWithManglingTask : compileBuildWithoutManglingTask,
 				cleanExtensionsBuildTask,
 				compileNonNativeExtensionsBuildTask,
 				compileCopilotExtensionBuildTask,
 				compileExtensionMediaBuildTask,
 				minified ? minifyVSCodeTask : bundleVSCodeTask,
-				vscodeTaskCI
+			));
+			gulp.task(prepackTask);
+			vscodeTask = task.define(`vscode${dashed(platform)}${dashed(arch)}${dashed(minified)}`, task.series(
+				prepackTask,
+				packingTask,
 			));
 		}
 		gulp.task(vscodeTask);
